metadata-ingestion/docs/sources/kafka/kafka.md

```diff
@@ -241,3 +241,358 @@ config:
 ```
 
 The underlying implementation is similar to [dbt meta mapping](https://docs.datahub.com/docs/generated/ingestion/sources/dbt#dbt-meta-automated-mappings), which has more detailed examples that can be used for reference.
+
+### Schema Resolution
+
+DataHub provides comprehensive schema resolution for Kafka topics with multiple fallback strategies. This feature is **completely independent** of data profiling and specifically handles cases where schemas cannot be found in the schema registry.
+
+**Key Points:**
+
+- Schema resolution works by sampling messages from Kafka topics to extract record names and infer schemas
+- It can be enabled/disabled independently of data profiling
+- It's primarily for **metadata extraction** (schema discovery), not data quality analysis
+- Data profiling (covered in the next section) is for **statistical analysis** of message content
+
+#### Multi-Stage Schema Resolution
+
+When DataHub cannot find schemas in the schema registry, it uses a comprehensive multi-stage approach to resolve schemas for Kafka topics:
+
+```yaml
+source:
+  type: kafka
+  config:
+    # ... other config ...
+
+    # Configure comprehensive schema resolution
+    schema_resolution:
+      enabled: true # Enable multi-stage schema resolution (disabled by default)
+      sample_timeout_seconds: 2.0 # Timeout per topic for record name extraction and schema inference
+      sample_strategy: "hybrid" # "earliest", "latest", or "hybrid"
+      max_messages_per_topic: 10 # Maximum messages to sample per topic (default: 10)
+
+    profiling:
+      max_workers: 20 # Controls parallelization for both profiling and schema resolution (default: 5 * CPU cores)
+      nested_field_max_depth: 5 # Maximum recursion depth for nested JSON structures (default: 10)
+```
+
+When `schema_resolution.enabled` is true (or the legacy `schemaless_fallback.enabled`):
+
+DataHub uses a comprehensive multi-stage approach to resolve schemas for Kafka topics:
+
+1. **TopicNameStrategy**: Direct lookup using `<topic>-key/value` pattern (most common)
+2. **TopicSubjectMap**: User-defined topic-to-subject mappings via `topic_subject_map` config
+3. **RecordNameStrategy**: Extract record names from message data and lookup `<record_name>-key/value`
+4. **TopicRecordNameStrategy**: Combine topic + record name: `<topic>-<record_name>-key/value`
+5. **Schema Inference**: As final fallback, infer schema from message data analysis
```

Comment:
Perfectly explained! ❤️ 

It will be great to have these funnel properly tracked in the reporting metrics.